File: isma-lisma/src/main/java/ru/nstu/isma/in/lisma/analysis/parser/LismaErrorListener.java
package ru.nstu.isma.in.lisma.analysis.parser;

import error.IsmaError;
import error.IsmaErrorList;
import org.antlr.v4.runtime.ANTLRErrorListener;
import org.antlr.v4.runtime.Parser;
import org.antlr.v4.runtime.RecognitionException;
import org.antlr.v4.runtime.Recognizer;
import org.antlr.v4.runtime.atn.ATNConfigSet;
import org.antlr.v4.runtime.dfa.DFA;

import java.util.BitSet;

/**
 * Created by Bessonov Alex
 * Date: 04.01.14
 * Time: 22:01
 */
public class LismaErrorListener implements ANTLRErrorListener {
    private IsmaErrorList errors;

    public LismaErrorListener(IsmaErrorList errors) {
        this.errors = errors;
    }

    @Override
    public void syntaxError(Recognizer<?, ?> recognizer, Object o, int i, int i2, String s, RecognitionException e) {
        IsmaError err = new IsmaError(i, i2, s);
        err.setType(IsmaError.Type.SYNTAX);
        errors.add(err);
    }

    @Override
    public void reportAmbiguity(Parser parser, DFA dfa, int i, int i2, boolean b, BitSet bitSet, ATNConfigSet atnConfigs) {
    }

    @Override
    public void reportAttemptingFullContext(Parser parser, DFA dfa, int i, int i2, BitSet bitSet, ATNConfigSet atnConfigs) {
    }

    @Override
    public void reportContextSensitivity(Parser parser, DFA dfa, int i, int i2, int i3, ATNConfigSet atnConfigs) {
    }
}

File: isma-lisma/src/main/kotlin/ru/nstu/isma/lisma/parser/LismaErrorListener.kt
package ru.nstu.isma.lisma.parser

import ru.nstu.isma.core.hsm.models.IsmaError
import org.antlr.v4.runtime.ANTLRErrorListener
import org.antlr.v4.runtime.Parser
import org.antlr.v4.runtime.RecognitionException
import org.antlr.v4.runtime.Recognizer
import org.antlr.v4.runtime.dfa.DFA
import org.antlr.v4.runtime.atn.ATNConfigSet
import ru.nstu.isma.core.hsm.models.IsmaSyntaxError
import java.util.*

/**
 * Created by Bessonov Alex
 * Date: 04.01.14
 * Time: 22:01
 */
class LismaErrorListener(private val errors: MutableList<IsmaError>) : ANTLRErrorListener {
    override fun syntaxError(
        recognizer: Recognizer<*, *>?,
        o: Any,
        i: Int,
        i2: Int,
        s: String,
        e: RecognitionException
    ) {
        val err = IsmaSyntaxError(i, i2, s)
        errors.add(err)
    }

    override fun reportAmbiguity(
        parser: Parser,
        dfa: DFA,
        i: Int,
        i2: Int,
        b: Boolean,
        bitSet: BitSet,
        atnConfigs: ATNConfigSet
    ) {
    }

    override fun reportAttemptingFullContext(
        parser: Parser,
        dfa: DFA,
        i: Int,
        i2: Int,
        bitSet: BitSet,
        atnConfigs: ATNConfigSet
    ) {
    }

    override fun reportContextSensitivity(
        parser: Parser,
        dfa: DFA,
        i: Int,
        i2: Int,
        i3: Int,
        atnConfigs: ATNConfigSet
    ) {
    }
}